### ISSUE TITLE
verbose html when enabling sanitizer

### ISSUE THREAD

**Issue by Eraldo** [dimpu/angular2-markdown#71]
I enabled the  sanitizer:
```
this.markdownService.setMarkedOptions({gfm: true, breaks: true, sanitize: true});
```
Now when I render my page, I get plain html including the html tags as text.
Taken from chrome inspector:
```
<markdown _nghost-c0="" ng-reflect-data="![day image](https://static01."><p>&lt;p&gt;&lt;img src="https://static01.nyt.com/images/2013/03/26/health/26well_physed/26well_physed-tmagArticle.jpg" alt="day image"&gt;&lt;/p&gt;<br>&lt;p&gt;Foo Bar new&lt;br&gt;asdfasdf Bla&lt;br&gt;asdfasdf&lt;/p&gt;<br>&lt;h2 id="bla"&gt;Bla&lt;/h2&gt;<br>&lt;p&gt;foo&lt;/p&gt;<br>&lt;p&gt;&lt;strong&gt;bold space&lt;/strong&gt;&lt;/p&gt;<br>&lt;p&gt;asdfasdf Ero&lt;/p&gt;<br>&lt;p&gt;My Image&lt;/p&gt;
</p>
</markdown>
```
In the browser it looks like this:
```html
<p><img src="https://static01.nyt.com/images/2013/03/26/health/26well_physed/26well_physed-tmagArticle.jpg" alt="day image"></p>
<p>Foo Bar new<br>asdfasdf Bla<br>asdfasdf</p>
<h2 id="bla">Bla</h2>
<p>foo</p>
<p><strong>bold space</strong></p>
<p>asdfasdf Ero</p>
<p>My Image</p>
```
What I expected was this:

----
![day image](https://static01.nyt.com/images/2013/03/26/health/26well_physed/26well_physed-tmagArticle.jpg)

Foo Bar new
asdfasdf Bla
asdfasdf

## Bla
foo

**bold space**

asdfasdf Ero

My Image

----

In other words, the sanitizer is screwing up the output so that I end up with html code instead of presented html.

This is how I bind to the component: 
```
  <div class="journal" padding *ngIf="entry$ | async as entry">
    <markdown [data]="entry.content"></markdown>
  </div>
```
The strange thing is: As soon as I emit another value for the observable which I use as a source everything does display as expected!
What is going on here?

**Comment by Eraldo** [comment]
I guess having a proper configuration option would solve this issue.
My suspicion is that it has something to do with timing due to both the options and the display happening in the same component.

**Comment by dimpu** [comment]
@Eraldo could you tell me what's your expected input and output. If possible do plunker plz?

**Comment by Eraldo** [comment]
I tried to add input and output in my issue description.
Which part is unclear?

**Comment by martin-watts** [comment]
It looks like the problem is that if marked 'sanitize' is set to true and you're using the `data` property binding:

1) In `onDataChange` the innerHTML is set properly, then
2) After init `processRaw` is called, and the innerHTML is recompiled (presumably to cope with the raw markdown being bound as the tag content in the template) but the sanitize option escapes HTML, so the innerHTML of `this.el.nativeElement.innerHTML` gets set to the 'raw' HTML instead.


The fix is to not call `processRaw` after init if the markdown was provided via the `data` property.


Setting the markdown using the tag content and curly bracket binding kind-of works, but Angular escapes the content, making things like blockquotes impossible.

**Comment by martin-watts** [comment]
Ah ok, I see this is fixed in github but the latest version available via NPM is '1.6.0' from '2017-04-23T08:46:30.309Z' which is before the fix went in.

@dimpu Any chance you can push a new version to npm?
